### DISCUSSION TITLE
"Best" SCSI Drivers for Macintosh?

### DISCUSSION THREAD

**Discussion by gingerbeardman** [erichelgeson/BlueSCSI-v1#90] [Q&A] [answered]
This page linked to from Performance https://github.com/erichelgeson/BlueSCSI/wiki/Disk-Drivers-(Mac) mostly lists which drivers to _not_ use.

But what is the best choice to use?
Does it differ by Macintosh/hardware type?

Results from my Macintosh Classic differ when using different drivers:

- (Apple) SC HD Setup 7.3.5
  > ![IMG_1562](https://user-images.githubusercontent.com/49612/163448806-ef5ccfe9-2254-4dce-ab11-eaca1b28459b.jpeg)

- (Transoft) SCSI Director Pro 4.0
  > ![IMG_1561](https://user-images.githubusercontent.com/49612/163448785-5f5ce6a9-3975-4036-bb93-ef10595b63b2.jpeg)

**Comment by erichelgeson** [comment, marked as answer]
It depends how you define best - if you're on a Mac Plus it's going to be different than if you are on a PowerMac. Also it depends on if you're shipping images between different computers a lot and want broader compatibility. 

For best compatibility I would say Apple SC HD Setup 7.3.5 is your best bet.

For best performance on a non-Plus system I would say MacOS 8 version of Drive Setup is the best (works on lower OS's as well, and the Plus, but oddly not with multiple drives on the Plus!)

Great thing is you can always update and test what works best for you.

General guidelines and "it depends" :)

**Reply by gingerbeardman** [reply]
I'm still waiting for a microSD to SD adapter cable to arrived to try high end regular SD cards.

But, so far, I am limited to a max of:
- ~560KB/s read
- ~440KB/s
- ~2.3ms seek

I have tested the following MicroSD cards
- SanDisk EVO 64GB class U1 i SDXC (numbers directly above) 
  - on macOS 12 this card 2X faster than the others below
- Kingston 8GB class 10 (very similar xfer numbers to above, but ~3ms seek)
- SanDisk 4GB class 4 (photos at top)
- Toshiba 4GB class 4
- SanDisk 4GB class 2

My setup
- 1992 Macintosh Classic 68K running System 7.1
- BlueSCSI running v1.1-20220404 bought from http://bluescsi.flamelily.co.uk
- SD cards are formatted to exFAT
- SCSI Director Pro 4 to benchmark

Formatters/Drivers
- HDD disk image formatted with Apple HD SC 7.3.5 (patched)
- HDD disk image formatted with SCSI Director Pro 4.0

Thoughts
- SCSI Director is best for my low end microSD
- Apple HD SC is best using my high end microSD

Question
- What am I missing here that I'm not getting anywhere close to 1.2MB/s?
- Am I at the limit of my Macintosh Classic?

SCSI2SD
- I get pretty much the same results (regular size SanDisk 4GB class 4 SDHX)
  - ~550KB/s
  - ~450KB/s
  - ~3ms seek

**Reply by erichelgeson** [reply]
@gingerbeardman These are the speeds I'd expect from a 8mhz 68k machine. 1.2mb/sec is achievable on fast 040's and PPC Machines.

**Reply by gingerbeardman** [reply]
> @gingerbeardman These are the speeds I'd expect from a 8mhz 68k machine. 

Excellent! My journey is over. 

Thanks for your hard work on this. It rocks!

**Reply by gingerbeardman** [reply]
I [read on lowendmac](https://lowendmac.com/1998/scsi-throughput/) that the limit for SCSI on 68K Mac is *5,248 Kbps (roughly 0.65 MBps)* so we're pretty close. Though they neglected to say if that was the theoretical or practical limit.

**Comment by gingerbeardman** [comment]
What version of Drive Setup came with Mac OS 8? I run 7 and 9 but not 8 or 8.5

**Reply by erichelgeson** [reply]
2.0.7? or maybe try 2.1? https://macintoshgarden.org/apps/drive-setup-21 Later drive setups do require a later OS. The details I dont know off the top of my head.

**Reply by JDW1** [reply]
When trying to launch version 2.1 on System 7.5, I get this error:
<img width="767" alt="image" src="https://user-images.githubusercontent.com/3928027/163717454-11754b63-94eb-464a-8dc6-b67557d894c6.png">

**Reply by gingerbeardman** [reply]
IIRC -192 error is trying to run System 8/8.5/9 (PPC?) app on System 7 (68K).

Maybe it would possible to write the driver to the disk whilst in System 8/9, then boot back to System 7?
